### ISSUE TITLE
SingleVariableDeclarationEnhancedForRenameParameterToCent bug

### ISSUE THREAD

**Issue by OriRoth** [SpartanRefactoring/Spartanizer#1120]
This tipper **sometimes** does not change the names of the variable occurrence's inside the enhanced for loop:
```Java
for (X x : xs) {
  ... x ...
}
```
==>
```Java
for (X ¢ : xs) {
  ... x ...
}
```
I suspect this is because the *domain* of this tipper is `SingleVariableDeclaration`, and not `EnhancedForStatement`. This makes its statement son applicable for spartanization, thus our changes get overriden by another tipper! I could not recreate the problem, though it appears multiple times while trying to spartanize large projects (as JUnit).
Also, why do we need two tippers for this task? See `EnhancedForParameterRenameToCent`.

**Comment by OriRoth** [comment]
@yossigil

**Comment by OriRoth** [comment]
A possible solution would be to stop consolidating tips on the current sub-tree when applying spartanization (a good behavior, in my opinion).

**Comment by yossigil** [comment]
I thought I solved it.

**Comment by yossigil** [comment]
Working on it now

**Comment by OriRoth** [comment]
I truly believe a general solution is needed here. If I am right about the spartanization override, the use of the global excluder is needed (or changing the tipper's domain to `EnhancedForStatement`).

**Comment by yossigil** [comment]
You are right. Here is how I see it. We have a class that is supposed to be doing all this. The `Inliner`.
Not sure what's it status.  But, it used to take a name, and a node or several nodes. Then, it would tell you 
if it is OK to inline into these nodes, and if possible, will do that inlining for you.

**Comment by OriRoth** [comment]
Fixed in 86b8b17d30b64906aa8164e1b07bf5ee73342433
